### DISCUSSION TITLE
Suddenly not working (Fedora 37, Gnome 43.2)

### DISCUSSION THREAD

**Discussion by junadlao** [neffo/bing-wallpaper-gnome-extension#197] [Q&A] [answered]
Hi. I don't wanna open an Issue yet as it might resolve itself but this has been working every day up to now. Wondering if anybody else is experiencing the same on Fedora 37, Gnome 43.2.

**Comment by neffo** [comment]
Thanks for report, I'll spin up a VM with latest updates and see if I can reproduce.

**Comment by neffo** [comment]
Hmm, I don't know if I can replicate. Can you explain more about what you are seeing?

**Comment by junadlao** [comment]
Maybe just spin out a quick 15 GB VM and install from Fedora's website their standard Workstation full installation with all the default settings and update all packages. That's how I made my system, but recently I upgraded to Fedora 37 (comes with Gnome 43.2) but that was 2 months ago but as I said this couldn't be it because the extension was working fine everyday -- I know 'coz I look at the different wallpapers everyday (Thanks to you, BTW ;)). Up until yesterday when it stopped. I tried removing and re-installing from the GNOME Shell extensions website but it just says "ERROR" with no further explanation. Please let me know if you need logs or anything else and where I can find them. Tnx.

**Comment by neffo** [comment]
Oh, sorry yeah I have a Fedora 37 VM I use for testing. I booted it up but I didn't see any problems. Are you able to open the preferences?

If you are enable debug logging (top-most toggle) you should see some additional output in the journal, but you would also see any errors that are triggered when it starts up.

![image](https://user-images.githubusercontent.com/5569112/217948464-f98478bc-68e4-4eb0-a94b-e6c8122f4c75.png)

You can take a look at the journal with:
```
journalctl --user -e
```

Though, presumably it isn't getting far. You should see something like this on a successful refresh:

```
Feb 10 08:02:06 ubuntu-desktop-22 gnome-shell[3289]: BingWallpaper extension: next check in 86400 seconds @ local time 2023-02-11 08:02 +1000
Feb 10 08:02:06 ubuntu-desktop-22 gnome-shell[3289]: BingWallpaper extension: Recieved 5180 bytes
Feb 10 08:02:06 ubuntu-desktop-22 gnome-shell[3289]: BingWallpaper extension: Next refresh due 36173.322369 seconds from now
Feb 10 08:02:06 ubuntu-desktop-22 gnome-shell[3289]: BingWallpaper extension: next check in 36173.322369 seconds @ local time 2023-02-10 18:05 +1000
Feb 10 08:02:06 ubuntu-desktop-22 gnome-shell[3289]: BingWallpaper extension: getCurrentImageIndex for 202302090800: 557
Feb 10 08:02:06 ubuntu-desktop-22 gnome-shell[3289]: BingWallpaper extension: getCurrentImageIndex for 202302090800: 557
Feb 10 08:02:06 ubuntu-desktop-22 gnome-shell[3289]: BingWallpaper extension: scale factor: 1
Feb 10 08:02:06 ubuntu-desktop-22 gnome-shell[3289]: BingWallpaper extension: Storing state as JSON: {"maxlongdate":202302090800,"title":"Lower Antelope Canyon, Arizona, USA","explanation":"Bing Wallpaper of the Day for 2023-02-09","copyright":"© AZCat/Getty Images","longstartdate":"202302090800","imageinfolink":"https://www.bing.com/search?q=Lower+Antelope+Canyon+Arizona&form=hpcapt&filters=HpDate%3a%2220230209_0800%22","imageURL":"https://www.bing.com/th?id=OHR.LowerAntelopeAZ_ROW4220166725_UHD.jpg","filename":"/mnt/2t_btr1/BingWallpaper/20230209-LowerAntelopeAZ_ROW4220166725_UHD.jpg","favourite":false}
```

**Comment by neffo** [comment, marked as answer]
![remmina_oldman local_oldman local_20230209-221437](https://user-images.githubusercontent.com/5569112/217951456-bb8a7657-b889-4511-8626-30272c4997d8.png)

**Reply by junadlao** [reply]
So, when I moved to a different computer with similar setup, your extension was working there as ever with no problems. When I moved back to the original computer, I first logged in to Gnome X11 and the extension was working (It was on Wayland originally where I got the problem). It was also working when I logged back in to Wayland which I prefer coz Wayland seems faster to me in general. So it did resolve itself and I don't know how or why.
I'm marking your reply as an answer and you have been very patient with this. I appreciate that.

**Reply by neffo** [reply]
Thanks, glad you enjoy it. I'm not aware of any issues with Wayland atm, but will keep an eye out.
